### PR TITLE
Support spring relaxed binding names, #538.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Only keep one record in serects map.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: azure-spring-boot/src/test/java/com/microsoft/azure/keyvault/spring/KeyVaultOperationUnitTest.java [hunk 1]
```diff
@@ -63,17 +76,28 @@ public void setup() {
...
+
         final String[] result = keyVaultOperation.list();
 
         assertThat(result.length).isEqualTo(1);
-        assertThat(result[0]).isEqualTo(testPropertyName1);
+        assertThat(result[0]).isEqualToIgnoringCase(testPropertyName1);
+    }
+
+    @Test
+    public void setTestSpringRelaxedBindingNames() {
```
[the remark this answers is not in the document]
Comment (Incarnation-p-lee) [reply azure-spring-boot/src/test/java/com/microsoft/azure/keyvault/spring/KeyVaultOperationUnitTest.java]: We can also leverage other assertion package, will disable this rule for this repo.

File: azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java [hunk 2]
```diff
@@ -33,68 +37,84 @@ public KeyVaultOperation(final KeyVaultClient keyVaultClient, final String vault
...
     }
 
-    public Object get(final String secretName) {
-        // NOTE: azure keyvault secret name convention: ^[0-9a-zA-Z-]+$ "." is not allowed
-        final String localSecretName = secretName.replace(".", "-");
+    private String getKeyvaultSecretName(@NonNull String property) {
+        if (property.matches("[a-z0-9A-Z-]+")) {
+            return property.toLowerCase(Locale.US);
+        } else if (property.matches("[A-Z0-9_]+")) {
+            return property.toLowerCase(Locale.US).replaceAll("_", "-");
```
Comment (sophiaso) [inline azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java]: Not sure whether using regex match is too heavy here.

Comment (sophiaso) [reply azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java]: `[A-Z0-9_]+` why the lowercased characters not included in the regexp?

File: azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java [hunk 3]
```diff
@@ -33,68 +37,84 @@ public KeyVaultOperation(final KeyVaultClient keyVaultClient, final String vault
...
+     * <table>
+     * <tr><td>Spring relaxed binding names</td></tr>
+     * <tr><td>acme.my-project.person.first-name</td></tr>
+     * <tr><td>acme.myProject.person.firstName</td></tr>
+     * <tr><td>acme.my_project.person.first_name</td></tr>
+     * <tr><td>ACME_MYPROJECT_PERSON_FIRSTNAME</td></tr>
+     * </table>
+     * But azure keyvault only allows ^[0-9a-zA-Z-]+$ and case insensitive, so there must be some conversion
+     * between spring names and azure keyvault names.
+     * For example, the stated below 4 properties should be convert to acme-myproject-person-firstname in keyvault.
```
Comment (sophiaso) [inline azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java]: Comment line 79 move to up? There is no `below 4 properties`.

Comment (Incarnation-p-lee) [reply azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java]: Oh, typo, I mean above here, will fix it ASAP.

**Pull request merged**